<!DOCTYPE html>
<html>
<body>
<div id="demo">
    <h1>Utilizando o Fetch</h1>
    <input type="text" name="cep" id="cep" placeholder="Digite seu CEP">
    <button type="button" onclick="buscaCEP()">Pesquisar CEP</button>
    <br>
    <input type="text" id="logradouro" placeholder="Logradouro">
    <br>
    <input type="text" id="complemento" placeholder="Complemento">
    <br>
    <input type="text" id="unidade" placeholder="Unidade">
    <br>
    <input type="text" id="bairro" placeholder="Bairro">
    <br>
    <input type="text" id="localidade" placeholder="Cidade">
    <br>
    <input type="text" id="uf" placeholder="UF">
</div>
<script>
function buscaCEP() {
  // obtém o CEP digitado - não faz nenhuma validação
  // para simplificar vamos acreditar que o usuário informará um CEP válido
  let cep =  parseInt(document.getElementById("cep").value); 

  const url = "https://viacep.com.br/ws/"+cep+"/json"

  // O fetch retorna uma Promise com o objeto Response
  fetch(url)

  // O primeiro .then() lida com o objeto Response
  .then(response => {
    // É crucial verificar se a resposta HTTP foi bem-sucedida (status 200-299)
    if (!response.ok) {
      // Se não for 'ok' (ex: 404, 500), lançamos um erro
      throw new Error(`Erro HTTP! Status: ${response.status}`);
    }

    console.log('resposta recebida - status OK');

    // Converte o corpo da resposta para o formato desejado (ex: JSON)
    return response.json(); 
  })

  // O segundo .then() lida com os dados já convertidos (o objeto JSON)
  .then(endereco => {
    console.log('Dados recebidos:', endereco);
   
    // Aqui você trabalha com os dados no seu código

    document.getElementById("logradouro").value=endereco.logradouro;
    document.getElementById("complemento").value=endereco.complemento ;
    document.getElementById("unidade").value=endereco.unidade;
      
    document.getElementById("bairro").value=endereco.bairro ;
    document.getElementById("localidade").value=endereco.localidade ;
    document.getElementById("uf").value=endereco.uf ;
  })
  // .catch() lida com erros de rede ou o erro lançado no primeiro .then
  .catch(error => {
    console.error('Ocorreu um erro na requisição:', error);
  });
}
</script>

</body>
</html>